web_src/js/features/gitgraph.js

```diff
@@ -46,6 +46,67 @@ export default async function initGitGraph() {
       window.history.replaceState({}, '', window.location.pathname);
     }
   });
+  const url = new URL(window.location);
+  const params = url.searchParams;
```

Comment:
Can get the same from `window.location.search`so I think you can remove the `url` variable completely.

Reply:
Actually I noticed you query params below so it will not work with a string.

Reply:
I do not understand what you mean.

The reason why I have const url and params is because I want a single params object to update so that when update is called it always sends the up-to-date info. That is, if you quickly select two branches before the ajax request returns for the first branch the second ajax query will still query for both branches.

If you do not have a global params object and recreate from the window.location each time you're going to end up with the situation that the second ajax query will not request for the first branch too.

Reply:
I guess it's fine, I was only commenting on the URL parsing but I later noticed you are using `url` below.